Sources/Fluent/Query/QueryRepresentable.swift

```diff
@@ -107,6 +107,7 @@ extension QueryRepresentable where Self: ExecutorRepresentable {
 
         if let _ = entity.id, entity.exists {
             // update
+            try type(of: entity).willUpdate(entity: entity)
```

Comment:
does `Self` work here as opposed to `type(of:`? I'm not sure, but it feels like it might be a little smoother process to not be doing a runtime type lookup when we can get it at compile. This would apply to all instances.